addons/binding/org.openhab.binding.dscalarm/src/main/java/org/openhab/binding/dscalarm/handler/DSCAlarmBaseThingHandler.java

```diff
@@ -0,0 +1,350 @@
+/**
+ * Copyright (c) 2014-2015 openHAB UG (haftungsbeschraenkt) and others.
+ *
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ */
+package org.openhab.binding.dscalarm.handler;
+
+import java.util.EventObject;
+import java.util.List;
+
+import org.eclipse.smarthome.core.thing.Bridge;
+import org.eclipse.smarthome.core.thing.Channel;
+import org.eclipse.smarthome.core.thing.ChannelUID;
+import org.eclipse.smarthome.core.thing.Thing;
+import org.eclipse.smarthome.core.thing.ThingStatus;
+import org.eclipse.smarthome.core.thing.binding.BaseThingHandler;
+import org.eclipse.smarthome.core.thing.binding.ThingHandler;
+import org.eclipse.smarthome.core.types.Command;
+import org.openhab.binding.dscalarm.config.DSCAlarmPanelConfiguration;
+import org.openhab.binding.dscalarm.config.DSCAlarmPartitionConfiguration;
+import org.openhab.binding.dscalarm.config.DSCAlarmZoneConfiguration;
+import org.openhab.binding.dscalarm.internal.DSCAlarmProperties;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * Abstract class for a DSC Alarm Thing Handler.
+ * 
+ * @author Russell Stephens - Initial Contribution
+ */
+public abstract class DSCAlarmBaseThingHandler extends BaseThingHandler {
+
+	private Logger logger = LoggerFactory.getLogger(DSCAlarmBaseThingHandler.class);
+	
+	/** Bridge Handler for the Thing. */
+	public DSCAlarmBaseBridgeHandler dscAlarmBridgeHandler = null;
+
+	/** DSC Alarm Thing type. */
+	private DSCAlarmThingType dscAlarmThingType = null;
+	
+	/** DSC Alarm Properties. */
+	public DSCAlarmProperties properties = null;
+	
+	/** This refresh status. */
+	private boolean thingRefreshed = false;
+
+	/** User Code for some DSC Alarm commands. */
+	private String userCode = null;
+	
+	/** Suppress Acknowledge messages when received. */
+	private boolean suppressAcknowledgementMsgs = false;
+
+	/** Partition Number. */
+	private int partitionNumber;
+
+	/** Zone Number. */
+	private int zoneNumber;
+
+	/**
+	 * Constructor.
+	 * 
+	 * @param thing
+	 */
+	public DSCAlarmBaseThingHandler(Thing thing) {
+		super(thing);
+	}
+
+	/**
+	 * {@inheritDoc}
+	 */
+	@Override
+	public void initialize() {
+		logger.debug("Initializing DSC Alarm Thing handler - Thing Type: {}; Thing ID: {}.", dscAlarmThingType, this.getThing().getUID());
+
+		getConfiguration(dscAlarmThingType);				
+		updateStatus(ThingStatus.OFFLINE);
+	}
+	
+	/**
+	 * {@inheritDoc}
+	 */
+	@Override
+	public void dispose() {
+		logger.debug("Thing {} disposed.", getThing().getUID());
+
+		this.setThingRefreshed(false);
+		super.dispose();
+	}
+
+	/**
+	 * Method to Refresh Thing Handler.
+	 */
+	public void refreshThing() {
+
+		if(getDSCAlarmBridgeHandler() != null) {
+			updateStatus(dscAlarmBridgeHandler.getThing().getStatus());
```

Comment:
you can imho remove this. You get calls on bridge connect and disconnect and therefore you can set the status there, which is enough.